### ISSUE TITLE
When a window is blurred and focused, the mode switches to command mode

### ISSUE THREAD

**Issue by saneef** [NeoVintageous/NeoVintageous#776]
**Describe the bug**  
While in insert mode, if you go to another app and focus back the window you are working on, the command mode will be activated. [Video](https://d.pr/v/5gc8Na)

**To Reproduce**  
1. Activate insert mode
2. Focus another app
3. Focus earlier Sublime Text window
4. You won't be in the insert mode.

This becomes an irritant when working between multiple apps.

**Versions**  
Show the Sublime Text console log: `Menu > View > Show Console`. 
Example: `startup, version: 4107 osx x64 channel: stable`

**Comment by haze** [comment]
Can reproduce this, is there any known workarounds until an official fix is in place?

**Comment by gerardroche** [comment]
Yes, set `vintageous_reset_mode_when_switching_tabs` to false and it will preserve the mode on window and view changes. That said the setting probably has some unwanted side effects i.e a window blur is not the same as a view change so you may *want* the mode to change on view changes but not on window blur. As far as I'm aware there isn't an API in ST to support doing that. Setting `vintageous_reset_mode_when_switching_tabs` to false is currently the best way to do it. p.s. I have a fix coming shortly for https://github.com/NeoVintageous/NeoVintageous/issues/791
